src/main/java/com/breskul/bibernate/persistence/GenericDao.java

```diff
@@ -180,6 +182,53 @@ public <T> T save(T entity) {
     return entity;
   }
 
+  /**
+   * Deletes an entity from the database using its ID. This method finds the entity's ID field,
+   * constructs a DELETE SQL query, and executes it. The entity must not be null and must have a
+   * non-null ID.
+   *
+   * @param <T>    the type of the entity
+   * @param entity the entity to be deleted
+   * @return the deleted entity
+   * @throws EntityQueryException     if there's an error during deletion or if the entity or its ID
+   *                                  is null
+   * @throws IllegalArgumentException if the entity is null
+   */
+  public <T> T delete(T entity) {
+    requireNonNull(entity, "Entity should not be null.");
+    Class<?> cls = entity.getClass();
+    String tableName = getEntityTableName(cls);
+    Field idField = findEntityIdField(cls);
+    idField.setAccessible(true);
+    try {
+      Object idObject = idField.get(entity);
+      if (idObject == null) {
+        throw new EntityQueryException("Entity ID is null for [%s]".formatted(entity));
+      }
+      String id = String.valueOf(idObject);
```

Comment:
Do we need it cast to string? let's use the prepared statement with question mark and set parameter for statement, it also help us avid sql injection